addons/udes_stock/models/stock_picking.py

```diff
@@ -2412,6 +2412,14 @@ def _reserve_stock_assign(self):
         moves = self.mapped("move_lines")
         moves.with_context(lock_batch_state=True)._action_assign()
 
+        # Unreserve any partially reserved lines if not allowed by the picking type
+        partial_moves = moves.filtered(lambda m: m.state == "partially_available")
+        moves_to_unreserve = partial_moves.filtered(
+            lambda m: m.picking_type_id.u_handle_partials
+            and not m.picking_type_id.u_handle_partial_lines
```

Comment:
Would it not read `u_handle_partial_lines` config on every line. Can we not save it before and then compare.

Reply:
Good point: I've changed this to group by picking type to avoid so many comparisions